app/assets/javascripts/analytics.js.erb

```diff
@@ -2,6 +2,28 @@
 var element = document.querySelector('[data-module="cookie-banner"]')
 new GOVUK.Modules.CookieBanner().start($(element))
 
+var referenceNumberPattern = function() {
+  var pattern = /\d{8}-\d{6}-.{6}/
+  return pattern
+}
+
+var productIdPattern = function() {
+  var pattern = /\b[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}\b/
+  return pattern
+}
+
+var stripPII = function(url) {
+  if (referenceNumberPattern().test(url)) {
+    var stripped = url.replace(referenceNumberPattern(), 'reference_number')
```

Comment:
Rather than setting stripped here, just return early e.g. `return url.replace(REF_NUMBER_PATTERN, 'reference_number')`. Then the `else` statement later becomes unnecessary and you can do `return url` at the end.

Reply:
👍 way better!